### ISSUE TITLE
Internal state display is garbled when too many sites are present

### ISSUE THREAD

**Issue by jkrivine** [Kappa-Dev/KaSim#408]
The new display for the contact map looks splendid! However the display of internal state is problematic when the kappa file gets big. For instance with the attached model.
Maybe the radius of the internal state circle should depend on the number of sites the model has ?

**Comment by vd1** [comment]
Nothing was attached as far as i can tell.
Where do you get that latest version with nice contact maps?
Vincent

**Comment by hmedina** [comment]
Determining draw radius by total number of sites might lead to a lot of
wasted space if the bulk of sites are on one agent, with singles elsewhere.
How about having a base radius per agent, each determined by the number of
sites in that agent, and an offset for the number of states the site has?
It might end up looking like a circular phylogeny tree...

**Comment by aznmonkey** [comment]
Can you link or upload the json file that you had issues with? I think I've found a solution but I'm not sure what type of state names are being used.

Also in general, what's the average length of the state names in terms of characters? Thanks!

**Comment by jkrivine** [comment]
I don't know why the file was not attached initially. Let me try again with the json file.
[contact.json.txt](https://github.com/Kappa-Dev/KaSim/files/1104946/contact.json.txt)

**Comment by aznmonkey** [comment]
I've partially fixed the issue. The length of the state lines now scale based on # of states present on the node (there is garbled text occassionally). Furthermore you can now hover the sites to see its states in the upper right hand corner.

**Comment by jkrivine** [comment]
The lenght of the line does not seem to adapt: the base_mod state has a 
lot of internal states and they are all superposed. Being able to see 
the list on the side is great though!

**Comment by aznmonkey** [comment]
I made it so that the lines extend out (of the viewbox) for sites with lots of states (the text is still garbled, this is just a temporary solution).  The hover window should be more readable now with the state links present. Due to the fact that there is really no limit on the number of states I'll have to think a bit more on a more permanent solution, but I think the current workaround should be fine.

I'm planning to work on snapshots/mixtures next, so the permanent fix for this might take a bit of time. Let me know if current workaround is too much of a hassle^^

**Comment by jkrivine** [comment]
This is already super cool. Thank you. I guess a solution is also not to display the internal states when they are two numerous and simply list them on the right as you did.

**Comment by aznmonkey** [comment]
I've made it so the text resize if they collide. Thus they will be much smaller, but still visible if you zoom in far enough.
